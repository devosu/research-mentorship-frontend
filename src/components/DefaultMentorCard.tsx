// @components/DefaultMentorCard
//
// DefaultMentorCard component definition.

// NextJS essential imports.

// Local imports.

interface Props {
    name: string
    major: string
    researchField: string
    researchDesc: string
    link: string
  }

export default function DefaultMentorCard (props: Props) {
return (
        <div className="bg-white shadow-md rounded-lg p-6">
            <h2 className="text-xl font-semibold">Name: {props.name}</h2>
            <p className="mt-2">Major: {props.major}</p>
            <p className="mt-2">Research Field: {props.researchField}</p>
            <p className="mt-2">Description of Research Field: {props.researchDesc}</p>
            <p className="mt-2">Outlook Scheduling Calendar Link: {props.link}</p>
        </div>
);
}