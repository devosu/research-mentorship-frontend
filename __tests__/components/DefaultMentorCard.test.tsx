// ./__tests__/components/DefaultMentorCard.test.tsx
//
// Snapshot tests for the DefaultMentorCard component.

// jsdom-jest testing essential imports.
import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";

// Local testing module imports.
import DefaultMentorCard, { type Props } from "@components/DefaultMentorCard";

// Behavior for the DefaultMentorCard component.
describe("The default mentor card component", () => {
  // Create a test prop to render and test the DefaultMentorCard component.
  const testProps: Props = {
    name: "Marison Bobson",
    major: "Electrical Engineering",
    researchField: "Power Systems",
    researchDesc: "I am currently researching power systems.",
    link: "https://outlook.com/schedule/marybobson",
  };

  it("renders a section element", () => {
    const { container } = render(
      <DefaultMentorCard
        name={testProps.name}
        major={testProps.major}
        researchField={testProps.researchField}
        researchDesc={testProps.researchDesc}
        link={testProps.link}
      />,
    );
    expect(container.querySelector("section")).toBeDefined();
  });

  it("matches the snapshot", () => {
    const { container } = render(
      <DefaultMentorCard
        name={testProps.name}
        major={testProps.major}
        researchField={testProps.researchField}
        researchDesc={testProps.researchDesc}
        link={testProps.link}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
