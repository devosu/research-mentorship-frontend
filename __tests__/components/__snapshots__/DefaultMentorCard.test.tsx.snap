// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The default mentor card component matches the snapshot 1`] = `
<div>
  <div
    class="bg-white shadow-md rounded-lg p-6"
  >
    <h2
      class="text-xl font-semibold"
    >
      Name: 
      Marison Bobson
    </h2>
    <p
      class="mt-2"
    >
      Major: 
      Electrical Engineering
    </p>
    <p
      class="mt-2"
    >
      Research Field: 
      Power Systems
    </p>
    <p
      class="mt-2"
    >
      Description of Research Field: 
      I am currently researching power systems.
    </p>
    <p
      class="mt-2"
    >
      Outlook Scheduling Calendar Link: 
      https://outlook.com/schedule/marybobson
    </p>
  </div>
</div>
`;
